plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt

```diff
@@ -243,15 +232,46 @@ fun reauthConnectionIfNeeded(
     project: Project?,
     connection: ToolkitConnection,
     onPendingToken: (InteractiveBearerTokenProvider) -> Unit = {},
+    isReAuth: Boolean = false
 ): BearerTokenProvider {
     val tokenProvider = (connection.getConnectionSettings() as TokenConnectionSettings).tokenProvider.delegate as BearerTokenProvider
     if (tokenProvider is InteractiveBearerTokenProvider) {
         onPendingToken(tokenProvider)
     }
 
     maybeReauthProviderIfNeeded(project, tokenProvider) {
-        runUnderProgressIfNeeded(project, AwsCoreBundle.message("credentials.pending.title"), true) {
-            tokenProvider.reauthenticate()
+        runUnderProgressIfNeeded(project, message("credentials.pending.title"), true) {
+            try {
+                tokenProvider.reauthenticate()
+                if (isReAuth) {
+                    recordLoginWithBrowser(
+                        credentialStartUrl = tokenProvider.currentToken()?.ssoUrl,
+                        credentialSourceId = CredentialSourceId.AwsId,
+                        isReAuth = true,
+                        result = Result.Succeeded
+                    )
+                    recordAddConnection(
+                        credentialSourceId = CredentialSourceId.AwsId,
+                        isReAuth = true,
+                        result = Result.Succeeded
+                    )
+                }
+            } catch (e: Exception) {
+                if (isReAuth) {
+                    recordLoginWithBrowser(
+                        credentialStartUrl = tokenProvider.currentToken()?.ssoUrl,
```

Comment:
but if we're passing along the start url, why do we need to also determine the source id?

Reply:
startUrl is required for locating internal users

Reply:
im saying that the source id is an implicit property of the start url, so why can't the BI team make this transform for us

Reply:
oh the other way around, got it. Yeah, doesn't add much value, but will let @dhruvigajjar comment

Reply:
no we dont need `credentialSourceId` if we will always have startUrl.